x-pack/plugins/fleet/server/services/agent_policies/package_policies_to_agent_inputs.ts

```diff
@@ -4,31 +4,56 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
+import type { SavedObjectsClientContract } from 'kibana/server';
 import { merge } from 'lodash';
 
-import type { PackagePolicy, FullAgentPolicyInput, FullAgentPolicyInputStream } from '../types';
-import { DEFAULT_OUTPUT } from '../constants';
+import { isPackageLimited } from '../../../common';
+import type { PackagePolicy, FullAgentPolicyInput, FullAgentPolicyInputStream } from '../../types';
+import { DEFAULT_OUTPUT } from '../../constants';
+
+import { getPackageInfo } from '../epm/packages';
 
-export const storedPackagePoliciesToAgentInputs = (
+export const storedPackagePoliciesToAgentInputs = async (
+  soClient: SavedObjectsClientContract,
   packagePolicies: PackagePolicy[],
   outputId: string = DEFAULT_OUTPUT.name
-): FullAgentPolicyInput[] => {
+): Promise<FullAgentPolicyInput[]> => {
   const fullInputs: FullAgentPolicyInput[] = [];
 
-  packagePolicies.forEach((packagePolicy) => {
+  for (const packagePolicy of packagePolicies) {
     if (!packagePolicy.enabled || !packagePolicy.inputs || !packagePolicy.inputs.length) {
-      return;
+      continue;
+    }
+
+    let appendInputId = true;
+
+    // Looks up package info and marks to skip appending input information to package policy ID
+    // to make it unique if package is "limited": this means that only one policy for the package can
+    // exist on the agent policy, so its ID is already unique
+    if (packagePolicy.package) {
+      const packageInfo = await getPackageInfo({
```

Comment:
`getPackageInfo` is a costly operation, you could pass from `calculateDiff`, where it is already queried.

Reply:
I split up this function into singular & plural ones, where the single version can accept a `packageInfo` from `calculateDiff`